uwh/xbee_comms.py

```diff
@@ -0,0 +1,80 @@
+from digi.xbee.devices import XBeeDevice
+from digi.xbee.exception import TimeoutException
+
+from . import messages_pb2
+from .comms import UWHProtoHandler
+
+import time
+
+class XBeeClient(UWHProtoHandler):
+    def __init__(self, mgr):
+        UWHProtoHandler.__init__(self, mgr)
+        self._xbee = XBeeDevice('/dev/tty.usbserial-DN040E8Y', 9600)
+        self._xbee.open()
+
+    def send_raw(self, recipient, data):
+        self._xbee.send_data_async(recipient, data)
+
+    def listen_loop(self):
+        while True:
+            xbee_msg = self._xbee.read_data()
+            if xbee_msg:
+                self.recv_raw(xbee_msg.remote_device, xbee_msg.data)
+                time.sleep(0.1)
+
+class XBeeServer(UWHProtoHandler):
```

Comment:
Not sure how/whether to write tests for these two classes. Is there a way to mark tests unsupported if specific hardware isn't available?

Reply:
(I did try to factor things such that a big portion of it is testable independantly of having actual hardware to test them on).